pkg/services/hcloud/server/server.go

```diff
@@ -405,53 +448,11 @@ func (s *Service) createServer(ctx context.Context) (*hcloud.Server, error) {
 		}
 	}
 
-	sshKeySpecs := s.scope.HCloudMachine.Spec.SSHKeys
-
-	// if no ssh keys are specified on the machine, take the ones from the cluster
-	if len(sshKeySpecs) == 0 {
-		sshKeySpecs = s.scope.HetznerCluster.Spec.SSHKeys.HCloud
-	}
-
-	// always add ssh key from secret if one is found
-
-	// this code is redundant with a similar one on cluster level but is necessary if ClusterClass is used
-	// as in ClusterClass we cannot store anything in HetznerCluster object
-	sshKeyName := s.scope.HetznerSecret().Data[s.scope.HetznerCluster.Spec.HetznerSecret.Key.SSHKey]
-	if len(sshKeyName) > 0 {
-		// Check if the SSH key name already exists
-		keyExists := false
-		for _, key := range sshKeySpecs {
-			if string(sshKeyName) == key.Name {
-				keyExists = true
-				break
-			}
-		}
-
-		// If the SSH key name doesn't exist, append it
-		if !keyExists {
-			sshKeySpecs = append(sshKeySpecs, infrav1.SSHKey{Name: string(sshKeyName)})
-		}
-	}
-
-	// get all ssh keys that are stored in HCloud API
-	sshKeysAPI, err := s.scope.HCloudClient.ListSSHKeys(ctx, hcloud.SSHKeyListOpts{})
+	caphSSHKeys, hcloudSSHKeys, err := s.getSSHKeys(ctx)
```

Comment:
this change is not documented in the PR description

Reply:
i added that to the PR description:

>  filterHCloudSSHKeys: The function does not filter, so it was renamed to getSSHKeys. Additionally, it was refactored to reduce the size of the createServer function.